text/utilities/0003-git-credential-configuration.md

```diff
@@ -0,0 +1,59 @@
+# Credential Configuration in the Git Buildpack
+
+## Summary
+
+Add the ability to supply credentials to the Git buildpack that will be used by
+`git` in order to allow later buildpacks to perform operations that require
+`git` authentication.
+
+## Motivation
+
+There are some operations, such as obtaining private language modules, that
+require `git` authentication during the build process. Currently there is no
+way of authenticating as a user to enable these workflows. Adding this
+functionality to the Git buildpack would open up a wider range of workflows
+than adding the functionality on an individual buildpack basis and make the
+ability to add credentials to the project universally.
+
+## Implementation
+
+For security purposes, the credentials cannot be provided through an
+environment variable like most of our other buildpack configuration as this
+poses a risk of exposure by view the build process in some form of CI. Because
+of this restriction, an approach similar to the CA Certificates Buildpack will
+be used where the credentials will be provided in a mounted volume through a
+service binding. Those credentials will then be used to change the global `git`
+config for that container which will persist for the entirety of the build
+which the [CNB specification](https://github.com/buildpacks/spec/blob/main/buildpack.md#requirements)
+states must all happen in the same container for security purposes.
```

Comment:
I read the referenced link, but I'm not seeing something that backs up this statement. It says:

>The lifecycle MUST be implemented so that the detection and build phases do not have access to OCI image store credentials used in the analysis and export phases. The lifecycle SHOULD be implemented so that each phase may run in a different container.

The first sentence there isn't relevant and I'm not sure I see the connection with the second sentence.

I'm not disagreeing. Do we have a better reference or explanation as to why this is OK?

Reply:
I was looking for a line in the spec that states that all of the build processes happen in one container and this was the best that I could come up with tbh. The point I am trying to get across is that I am making this setting for the build container and not for the final image therefore the credentials setting will only exist during build.

Reply:
What about with untrusted builders? https://buildpacks.io/docs/tools/pack/concepts/trusted_builders/#why-use-a-trusted-builder

Do we care about that scenario?

Reply:
I read that differently. I read it as each buildpack and phase would be separate. So if you had buildpack A and buildpack B, that would be separate containers for each phase for each buildpack for a total of four separate containers.

Reply:
All the build phases of every detected buildpack are run in one container.

Reply:
@samj1912 do you know if that is spec'd out somewhere

Reply:
Ok, I'm all set here. That clarifies it for me. If there is a link cool, if not, that's Ok too. Thanks